### PR TITLE
Feature/docker compose deploys

### PR DESCRIPTION
**Pull request by ashishmax31**
This PR adds the required files to enable docker-compose based production deployments.

### CODE REVIEW

File: docs/docs/deployment/docker.md [hunk 1]
```diff
@@ -1,5 +1,51 @@
...
+  :::info
+    If you're running on a linux server, `docker` might need sudo permissions. In that case you can either run:
+    `sudo docker-compose up -d`    
+    OR   
+    Setup docker to run without root privilages by following the instructions written here https://docs.docker.com/engine/install/linux-postinstall/ 
+  :::
+
+6.  If you've set a custom domain for `TOOLJET_HOST`, add a `A record` entry in your DNS settings to point to the IP address of the      server.
+
+7.  You're all done, ToolJet client would now be served at the value you've set in `TOOLJET_HOST`.
```
Comment (Navaneeth-pk) [inline docs/docs/deployment/docker.md:51]: `served at the value` should be `served at the URL`

File: docs/docs/deployment/docker.md [hunk 2]
```diff
@@ -1,5 +1,51 @@
 ---
-sidebar_position: 3
+sidebar_position: 2
+sidebar_label: Docker
 ---
+:::info
```
Comment (Navaneeth-pk) [inline docs/docs/deployment/docker.md:5]: This section should be below the title.

File: docs/docs/deployment/docker.md [hunk 3]
```diff
@@ -1,5 +1,51 @@
...
+Follow the steps below to deploy ToolJet ( both server & client ) on a server using docker-compose:
+
+1. Setup a PostgreSQL database and make sure that the database is accessible.
+
+2. Install docker and docker-compose on the server.  
+[Docker Installation](https://docs.docker.com/engine/install/)   
+[Docker Compose Installation](https://docs.docker.com/compose/install/)   
+
+3. Download our production docker-compose file into the server by running:
+`curl -LO https://raw.githubusercontent.com/ToolJet/ToolJet/develop/deploy/docker/docker-compose.yaml`
```
Comment (Navaneeth-pk) [inline docs/docs/deployment/docker.md:20]: The branch should be `main`

**Comment by ashishmax31** [comment]
@Navaneeth-pk Addressed the comments

**Pull request merged**